src/units.cpp

```diff
@@ -464,4 +464,99 @@ double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
     return 0.0;
 }
 
+void createUnitMap(const UnitsPtr &units, std::map<std::string, double> &unitMap, double exp = 1.0)
```

Comment:
Define a typemap for this `unitMap`; Something like `using UnitsMap = std::map<std::string, double>;`